public/controllers/agent/components/register-agent.js

```diff
@@ -963,11 +962,24 @@ export const RegisterAgent = withErrorBoundary(
         zIndex: '100',
       };
 
-       // Select macOS installation script based on architecture
-       const macOSInstallationOptions = (this.optionalDeploymentVariables() + this.agentNameVariable()).replaceAll('\' ', '\'\\n');
-       const macOSInstallationSetEnvVariablesScript = macOSInstallationOptions ? `echo -e "${macOSInstallationOptions}" > /tmp/wazuh_envs && ` : ``;
-       const macOSInstallationScript = `curl -so wazuh-agent.pkg https://packages.wazuh.com/4.x/macos/wazuh-agent-${this.state.wazuhVersion
-           }-1.pkg && ${macOSInstallationSetEnvVariablesScript}sudo installer -pkg ./wazuh-agent.pkg -target /`;
+      /*** macOS installation script customization ***/
+
+      // Set macOS installation script with environment variables
+      const macOSInstallationOptions = `${this.optionalDeploymentVariables()}${this.agentNameVariable()}`
+        .replace(/\' ([a-zA-Z])/g, '\' && $1') // Separate environment variables with &&
+        .replace(/\"/g, '\\"'); // Escape double quotes
+
+      // If no variables are set, the echo will be empty
+      const macOSInstallationSetEnvVariablesScript = macOSInstallationOptions ?
+        `sudo echo "${macOSInstallationOptions}" > /tmp/wazuh_envs && `
```

Comment:
In the example of this [comment](https://github.com/wazuh/wazuh-kibana-app/issues/5466#issuecomment-1559922195) it says `/tmp/wazuh_vars` and in the code it appears `/tmp/wazuh_envs`. Is it ok or does it need to be changed?

![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/40cc10fd-3d0d-4770-900a-dec6f3f02723)

Reply:
My bad @yenienserrano. Thanks for noticing

The proper and final filename is `/tmp/wazuh_envs`. @asteriscos was right.

This was clarified in the main issue description https://github.com/wazuh/wazuh-packages/issues/2205